src/drivers/upstash.ts

```diff
@@ -25,8 +25,9 @@ export default defineDriver<UpstashOptions, Redis>(
       if (redisClient) {
         return redisClient;
       }
-      redisClient = new Redis(options);
-
+      const url = options.url || process.env.UPSTASH_REDIS_REST_URL;
+      const token = options.token || process.env.UPSTASH_REDIS_REST_TOKEN;
```

Comment:
This would mean it won't work in environments without `process.env`

Reply:
Ah, looks like upstash/redis has a cross-platform static method `fromEnv`. Perhaps we could check if the options are defined, and if not then try to use that.